README.md

```diff
@@ -18,10 +18,198 @@ Or install it yourself as:
+
+Start the process under unicorn with `/usr/bin/litmus --unicorn-config <path_to_unicorn_conf>`. In the unicorn config file, set the number of worker processes, the pid, and the working directory. See the [unicorn documentation](https://bogomips.org/unicorn/Unicorn/Configurator.html) for the config format. There are also more options, see `/usr/bin/litmus -h` for a full list.
+
+For HAProxy agent checks, run `/usr/bin/litmus-agent-check`. See `/usr/bin/litmus-agent-check -h` for run options.
+
+### Global configuration
+
+Example:
+
+```
+# /etc/litmus.conf
+
+include_files "litmus.d/*.conf"
+
+port 80
+
+data_directory "/litmus"
+```
+
+Available fields:
+- include_files: Tells Litmus to load health check configurations from a path.
+- port: Port Litmus unicorn server will listen on, defaults to 9292.
+- data_directory: Where to store force down, up, and health files. Defaults to "/etc/litmus".
+- cache_location: Where to store cached health information, defaults to "/run/shm".
+- cache_ttl: Time to live in seconds for cached health check values, defaults to -1.
+
+### Service health check configuration
+
+To add services and health checks, Litmus Paper loads configurations written in ruby. Suppose you're writing a health check for a web application. You might start with a simple check to report if the server is responding at all:
+
+```ruby
+# /etc/litmus.d/myapp.conf
+service "myapp" do |s|
+  s.depends Dependency::HTTP, "http://localhost/heartbeat", :method => "GET", :ca_file => "/etc/ssl/certs/ca-certificates.crt"
+end
+```
+
+Maybe you also want to balance traffic based on CPU load:
+
+```ruby
+# /etc/litmus.d/myapp.conf
+service "myapp" do |s|
+  s.depends Dependency::HTTP, "http://localhost/heartbeat", :method => "GET", :ca_file => "/etc/ssl/certs/ca-certificates.crt"
+  s.measure_health Metric::CPULoad, :weight => 100
+end
+```
+
+Once you've finished adding checks, restart litmus paper to pick up the new service.
+
+Here are all the types of checks currently implemented:
+
+- `Dependency::HTTP`: Checks HTTP 200 response from a URL.
+  * url
+  * method (defaults to get)
+  * ca_file (defaults to nil)
+  * timeout (defaults to 2s)
+
+- `Dependency::TCP`: Checks successful completion of a TCP handshake with an address.
+  * ip
+  * port
+  * input_data (defaults to nil)
+  * expected_output (defaults to nil)
+  * timeout (defaults to 2s)
+
+- `Dependency::FileContents`: Checks whether the contents of a file match a string or regex.
+  * path
+  * regex
+  * timeout (defaults to 5s)
+
+- `Dependency::Script`: Checks whether the output of a command matches a string or regex.
+  * command
+  * timeout (defaults to 5s)
+
+- `Metric::ConstantMetric`: A dummy metric that always reports a constant.
+  * weight (0-100)
+
+- `Metric::CPULoad`: Normalizes CPU load to a value between 0-100 and inverts it, so higher numbers mean less load and lower numbers mean more. Final health is weighted against other checks by `:weight`.
+  * weight (0-100)
```

Comment:
Between 1-100. This is intentional, because the algorithm used to be `[(metric weight - (weight * load average / processor count)), 0].max`, and it was changed to `[(metric weight - (weight * load average / processor count)), 1].max` in https://github.com/braintree/litmus_paper/commit/4de5adf2af9f5a5c739feab130c6b6069fe6184d .

Consider the case where you've weighted 100, have 4 cpus, and have a load of 4:
`100 - (100 * 4/4) == 0`. Which takes your node out of the cluster. That then compounds load on the next node, and its load goes up ... and then the metric probably drops to zero ... and so on and so forth. So we now have a floor of `1` to prevent nodes from leaving the cluster solely on cpu health.

It's probably worth calling out the reason behind the behavior. A future improvement could be to make the floor configurable.